### PR TITLE
Rework UI input field

### PR DESCRIPTION
**Pull request by VladPodolianchik**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -33,7 +33,15 @@
         "react/react-in-jsx-scope": "off",
         "react/jsx-filename-extension": [1, { "extensions": [".tsx", ".jsx"] }],
         "react/jsx-wrap-multilines": ["error", {"declaration": "parens-new-line", "assignment": "parens-new-line", "return": "parens-new-line", "arrow": "parens-new-line", "condition": "parens-new-line", "logical": "parens-new-line", "prop": "parens-new-line"}],
-        "import/prefer-default-export": "off"
+        "import/prefer-default-export": "off",
+        "react/require-default-props": "off"
+    //     "react/function-component-definition": [
```
Comment (Lansyfonseka) [inline .eslintrc.json:38]: Why whis rules are commented? And why this rules are need?

File: src/UI/InputField/InputField.css [hunk 2]
```diff
@@ -1,18 +1,78 @@
...
+  /* margin just for test */
 }
 
-.input-field {
-  width: 60%;
-  height: 50px;
-  flex-grow: 1;
-  padding-right: 10px;
+.input-field-UI {
+  border: 2px solid gray;
```
Comment (Lansyfonseka) [inline src/UI/InputField/InputField.css:8]: You need use variables from src/styles/variables.scss.
Check this on another places such as color, font size, border radius, transition, @media and ...
Maybe you need new variables you can add

File: src/UI/InputField/InputField.tsx [hunk 3]
```diff
@@ -1,31 +1,70 @@
...
+import EyeIconHidden from '../../assets/icons/IconsSvg/EyeIconHidden';
 import './InputField.css';
 
-interface InputFieldType {
-  type: string;
-  id: string;
-  name: string;
-  minLength: number;
-  required: boolean;
+interface InputFieldProps {
```
Comment (Lansyfonseka) [inline src/UI/InputField/InputField.tsx:6]: Use style name like "IInputFieldProps" it's better. All interface start from short "I...."

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -3,12 +3,26 @@ import styles from './Header.module.scss';
...
 import { Path } from '../../pages/constants/path';
 import Navigation from '../Navigation/Navigation';
+// import InputField from '../../UI/InputField';
 
 export default function Header() {
   const { pathname } = useLocation();
 
   return (
     <header className={styles.header}>
+      {/* just for test */}
```
Comment (Lansyfonseka) [inline src/components/Header/Header.tsx:13]: This commented code don't needed. You can use this in your branch, but don't add in production version.

**Review by Lansyfonseka — COMMENTED** [review]
Now i can't test your branch in vscode. I try this later

**Comment by Lansyfonseka** [comment]
I think, need add area for error message, that UI elemenent not "jumping"
Eye icon is not center, try use "px"
Resolution <1130px (@madia) eye icon "run away", pls check this
image
You should write function thar return new value to parant element. (Don't use eslint-disable)

File: .eslintrc.json (hunk 1 above)
Comment (VladPodolianchik) [reply .eslintrc.json:38]: This is an optional rule. We can decide in which form we will write functions

File: src/UI/InputField/InputField.css (hunk 2 above)
Comment (VladPodolianchik) [reply src/UI/InputField/InputField.css]: fixed or at least tried to fix

File: .eslintrc.json (hunk 1 above)
Comment (AmidaWolf) [reply .eslintrc.json:38]: You can change or delete this rule. Never use comments for change rules. And always explain why rule should be changed/deleted.

File: src/UI/InputField/InputField.scss [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    margin-right: 8px;
+  }
+}
+
+.input-field-UI {
+  border: 2px solid $text-color-middle;
+  border-radius: $main-border-radius;
+  width: 100%;
+  height: 45px;
+  /* flex-grow: 1; */
```
Comment (AmidaWolf) [inline src/UI/InputField/InputField.scss:19]: Same thing about commented code. All commented code should be deleted or marked why this part of code should be commented.

File: src/UI/InputField/InputField.scss [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  color: red;
+}
+
+// @media (max-width: 1130px) {
+//   .input-field-UI {
+//     width: 100%;
+//     flex-grow: 1;
+//     padding: 10px;
+//   }
+// }
```
Comment (AmidaWolf) [inline src/UI/InputField/InputField.scss:81]: Same thing about commented code, as in prev message. All commented code should be deleted or marked why this part of code should be commented.

File: src/UI/InputField/InputField.scss [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+.input-group-UI {
+  position: relative;
+  margin-top: 20px;
+  .eye-icon {
+    position: absolute;
+    top: 21%;
+    right: 0;
+    margin-right: 8px;
+  }
+}
```
Comment (AmidaWolf) [inline src/UI/InputField/InputField.scss:12]: Let's imagine that the element has a height of 100 pixels. Then 21% is 21 pixels. But what if we change the content of the element or its dimensions? 21% becomes another distance. Do not rely on percentages to position elements that must be located at a specific and precise distance from the borders of this element.
Better yet, look at the main page of Google, at the search bar. You can make a form with all the icons without resorting to using absolute positioning. Or, if you can't do without it, use the hack with 50% and transform by -50%:
```
top: 50%;
transform: translate(0,-50%);
```
Also, eye is not just svg element - this is a button.

File: src/UI/InputField/InputField.tsx [hunk 8]
```diff
@@ -1,31 +1,70 @@
...
+  type?: string;
+  name?: string;
   placeholder: string;
+  className?: string;
+  errorText?: string;
+  isError?: boolean;
+  //   value?: string;
+  // eslint-disable-next-line no-unused-vars
+  onChange: (value: string) => void;
 }
```
Comment (AmidaWolf) [inline src/UI/InputField/InputField.tsx:16]: Please, place optional values after main values.

**Pull request closed without merging**